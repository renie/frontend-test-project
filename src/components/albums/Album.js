import React from 'react';
import './album.css'

const Album = ({album}) => {
  return (
    <li className="album">{album.title}</li>
  )
}

export default Album
