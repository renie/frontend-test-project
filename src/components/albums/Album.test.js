import React from 'react'
import ReactDOM from 'react-dom'
import Album from './Album'
import { shallow } from 'enzyme'

describe('Album', () => {
  it('Create Component', () => {
    const album = { title: 'Some pretty good title!'}
    const wrapper = shallow(<Album album={album} />)

    expect(wrapper).toMatchSnapshot()
  })
})
