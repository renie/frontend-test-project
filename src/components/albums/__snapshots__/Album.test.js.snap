// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album Create Component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Album
    album={
      Object {
        "title": "Some pretty good title!",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Some pretty good title!",
      "className": "album",
    },
    "ref": null,
    "rendered": "Some pretty good title!",
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Some pretty good title!",
        "className": "album",
      },
      "ref": null,
      "rendered": "Some pretty good title!",
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
